src/app/compliance/actions.js

```diff
@@ -199,3 +200,42 @@ export const removeInvestors = (addresses: Array<Address>) => async (dispatch: F
     true // TODO @bshevchenko
   ))
 }
+
+export const getFreezeStatus = () => async (dispatch: Function, getState: GetState) =>{
+  // $FlowFixMe
+  const transferManager = await getState().token.token.contract.getTransferManager()
+  transferManager.subscribePauseEvents((e)=>{
+    if(e.event==='pause'){
+      dispatch({ type: FREEZE_STATUS, freezeStatus: true })
+    }else if(e.event==='unpause'){
+      dispatch({ type: FREEZE_STATUS, freezeStatus: false })
+    }
+  })
+
+  dispatch({ type: FREEZE_STATUS, freezeStatus: await transferManager.paused() })
+}
+
+export const toggleFreeze = (postToggle: ?Function) =>
+  async (dispatch: Function, getState: GetState) =>{
+    const { freezeStatus } = getState().whitelist
+    // $FlowFixMe
+    const transferManager = await getState().token.token.contract.getTransferManager()
```

Comment:
you shouldn't request TransferManager second time, it was already requested in fetchWhitelist ac above, grab it from the state

Reply:
oh, you don't need transferManager here at all